### PR TITLE
[#127] Add notes stating the assumptions needed for the rules to work

### PR DESCRIPTION
**Pull request by elbrujohalcon**

### CODE REVIEW

File: src/rebar3_hank_prv.erl [hunk 1]
```diff
@@ -40,7 +40,7 @@ do(State) ->
     rebar_api:debug("Hank Context: ~p", [Context]),
     %% All files except those under _build or _checkouts
     Files = [F || F <- filelib:wildcard(?FILES_PATTERN), hd(F) /= $_],
-    rebar_api:debug("Hank will use ~p files for anlysis: ~p", [length(Files), Files]),
+    rebar_api:debug("Hank will use ~p files for analysis: ~p", [length(Files), Files]),
```
Comment (paulo-ferraz-oliveira) [inline src/rebar3_hank_prv.erl:43]: 👀

File: src/rules/single_use_hrl_attrs.erl [hunk 2]
```diff
@@ -4,6 +4,14 @@
 %%      -record
 %%      It will suggest to place those attributes inside the module to avoid
 %%      having (and including) a hrl file.
+%%
+%%      <h3>Note</h3>
+%%      <blockquote>
+%%      This rule assumes that hrl files will not be used outside your project.
+%%      If you are writing a library that require your clients to use some of
```
Comment (paulo-ferraz-oliveira) [inline src/rules/single_use_hrl_attrs.erl:11]:
```suggestion
%%      If you are writing a library that requires your clients to use some of
```

File: src/rules/single_use_hrls.erl [hunk 3]
```diff
@@ -1,6 +1,13 @@
 %% @doc A rule to detect header files used in just one module.
 %%      <p>To avoid this warning, include the content of the header file into
 %%      the module.</p>
+%%
+%%      <h3>Note</h3>
+%%      <blockquote>
+%%      This rule assumes that hrl files will not be used outside your project.
+%%      If you are writing a library that require your clients to use some of
```
Comment (paulo-ferraz-oliveira) [inline src/rules/single_use_hrls.erl:8]:
```suggestion
%%      If you are writing a library that requires your clients to use some of
```

File: src/rules/unused_hrls.erl [hunk 4]
```diff
@@ -1,5 +1,12 @@
 %% @doc A rule to detect unused header files.
 %%      <p>To avoid this warning, remove the unused header files.</p>
+%%
+%%      <h3>Note</h3>
+%%      <blockquote>
+%%      This rule assumes that hrl files will not be used outside your project.
+%%      If you are writing a library that require your clients to use some of
```
Comment (paulo-ferraz-oliveira) [inline src/rules/unused_hrls.erl:7]:
```suggestion
%%      If you are writing a library that requires your clients to use some of
```

File: src/rules/unused_macros.erl [hunk 5]
```diff
@@ -10,6 +10,14 @@
 %%      So, if you have a project with multiple definitions of the same macro
 %%      with the same arity... well... as long as one of them is used, none of
 %%      them will be reported as unused.
+%%
+%%      <h3>Note</h3>
+%%      <blockquote>
+%%      This rule assumes that hrl files will not be used outside your project.
+%%      If you are writing a library that require your clients to use a macro
```
Comment (paulo-ferraz-oliveira) [inline src/rules/unused_macros.erl:17]:
```suggestion
%%      If you are writing a library that requires your clients to use a macro
```

File: src/rebar3_hank_prv.erl (hunk 1 above)
Comment (elbrujohalcon) [reply src/rebar3_hank_prv.erl:43]: # 🤫

File: src/rules/unused_configuration_options.erl [hunk 6]
```diff
@@ -4,6 +4,16 @@
...
 %%      - The env list inside any *.app.src files
 %%      <p>To avoid this warning, remove the unused parameters.</p>
+%%
+%%      <h3>Note</h3>
+%%      <blockquote>
+%%      For this rule to apply, it's assumed that configuration options for an
+%%      Erlang application are only consumed within said Erlang application or
+%%      the other applications in the same umbrella project.
+%%      If you have a dependency that consumes an environment parameter from one
+%%      of your project applications, you can add an ignore rule in rebar.config
```
Comment (pbrudnick) [inline src/rules/unused_configuration_options.erl:14]: or refactor that 👀

**Review by pbrudnick — APPROVED** [review]

**Pull request merged**